frontend/test/pytest/test_gradient.py

```diff
@@ -688,5 +690,42 @@ def interpret_grad(params):
     assert np.allclose(compile_grad(jnp.array(inp)), interpret_grad(inp))
 
 
+def test_non_float_arg():
+    """Test a function which attempts to differentiate non-floating point arguments."""
+
+    @qml.qnode(qml.device("lightning.qubit", wires=2))
+    def circuit(x: complex, y: float):
+        qml.RX(jnp.real(x), wires=0)
+        qml.RY(y, wires=0)
+        return qml.expval(qml.PauliZ(0))
+
+    @qjit
+    def cost_fn(x, y):
+        return grad(circuit)(x, y)
+
+    with pytest.raises(
+        TypeError, match="only supports differentiation on floating-point arguments"
+    ):
+        cost_fn(1j, 2.0)
+
+
+def test_non_float_res():
+    """Test a function which attempts to differentiate non-floating point results."""
+
+    @qml.qnode(qml.device("lightning.qubit", wires=2))
+    def circuit(x: float, y: float):
+        qml.RX(x, wires=0)
+        qml.RY(y, wires=0)
+        return qml.expval(qml.PauliZ(0))
+
+    @qjit
+    @grad
```

Comment:
Weird use of the grad op as a decorator :)

Reply:
😆

Reply:
I sometimes do this, if you don't care about the original cost function and only need the grad